controllers/tutorial.controller.js

```diff
@@ -11,31 +12,29 @@ const getTutorials = async (req, res) => {
     res.status(200).send(tutorials);
   } catch (error) {
     res.status(500).send({ message: "Internal server error" });
+    logger.error('Internal server error');
   }
 };
 
 // Add new Tutorials
 const createTutorial = async (req, res) => {
-  const tutorial = new Tutorial({
-    title: req.body.title,
-    description: req.body.description,
-    published: req.body.published,
-  });
   try {
-    const { error } = postTutorialSchema.validate(req.body);
+    const { error, value } = postTutorialSchema.validate(req.body);
     if (error) {
-      res.status(400).send({ message: error.message });
-    } else {
-      await tutorial.save();
-      res.status(201).send();
+      return res.status(400).send({ message: error.message });
     }
+    const tutorial = new Tutorial(value);
+    await tutorial.save();
+    res.status(201).send();
   } catch (error) {
     if (error instanceof CastError) {
       res.status(400).send({ message: "Invalid Tutorial details" });
     } else {
       res.status(500).send({ message: "Internal Server Error" });
+      logger.error('Internal Server Error');
```

Comment:
We also need to log actual error, otherwise how would we know that what was the root cause of this 500?

Reply:
You can configure this method to print errors:

```
const winston = require('winston');
const { format } = require('logform');

const logger = winston.createLogger({
  format: format.combine(
    format.errors({ stack: true }),
    format.metadata(),
    format.json(),
  ),
  transports: [ new winston.transports.Console() ],
});
```
Now you can log errors like this:
```
logger.error(error);
```


For above code to work, You need to install `logform` dependency.

CC: @jaimin45 @JainishaParmar @ashishkava